### PR TITLE
api_core: Fix handling of gapic metadata when specified as 'None'.

### PR DESCRIPTION
**Pull request by jonparrott**

### CODE REVIEW

File: api_core/tests/unit/gapic/test_method.py [hunk 1]
```diff
@@ -94,6 +94,19 @@ def test_invoke_wrapped_method_with_metadata():
...
 
 
+def test_invoke_wrapped_method_with_metadata_as_none():
+    method = mock.Mock(spec=['__call__'])
+
+    wrapped_method = google.api_core.gapic_v1.method.wrap_method(method)
+
+    wrapped_method(mock.sentinel.request, metadata=None)
+
+    method.assert_called_once_with(mock.sentinel.request, metadata=mock.ANY)
```
Comment (dhermes) [inline api_core/tests/unit/gapic/test_method.py]: Why use `mock.ANY` here?

File: api_core/tests/unit/gapic/test_method.py [hunk 2]
```diff
@@ -94,6 +94,19 @@ def test_invoke_wrapped_method_with_metadata():
...
+    metadata = method.call_args[1]['metadata']
```
Comment (dhermes) [inline api_core/tests/unit/gapic/test_method.py]: Why are you checking `call_args`?

File: api_core/tests/unit/gapic/test_method.py (hunk 1 above)
Comment (jonparrott) [reply api_core/tests/unit/gapic/test_method.py]: Because I don't need the tests to re-calculate the metrics header.

Comment (dhermes) [reply api_core/tests/unit/gapic/test_method.py]: Ah

**Pull request merged**